pkg/target/target.go

```diff
@@ -330,20 +336,22 @@ func (h *K8sValidationTarget) MatchSchema() apiextensions.JSONSchemaProps {
 				Type: "array",
 				Items: &apiextensions.JSONSchemaPropsOrArray{
 					Schema: &apiextensions.JSONSchemaProps{
-						Type: "object",
+						Type:        "object",
+						Description: "The Group and Kind of objects that should be matched.  If multiple groups/kinds combinations are specified, an incoming resource need only match one to be in scope.",
 						Properties: map[string]apiextensions.JSONSchemaProps{
 							"apiGroups": nullableStringList,
 							"kinds":     nullableStringList,
 						},
 					},
 				},
 			},
-			"namespaces":         wildcardNSList,
-			"excludedNamespaces": wildcardNSList,
-			"labelSelector":      labelSelectorSchema,
-			"namespaceSelector":  labelSelectorSchema,
+			"namespaces":         *propsWithDescription(&wildcardNSList, "`namespaces` is a list of namespace names. If defined, a constraint will only apply to resources in a listed namespace.  Namespaces also supports a prefix-based glob.  For example, `namespaces: [kube-*]` would match both `kube-system` and `kube-public`."),
+			"excludedNamespaces": *propsWithDescription(&wildcardNSList, "`excludedNamespaces` is a list of namespace names. If defined, a constraint will only apply to resources not in a listed namespace. ExcludedNamespaces also supports a prefix-based glob.  For example, `excludedNamespaces: [kube-*]` would match both `kube-system` and `kube-public`."),
+			"labelSelector":      *propsWithDescription(&labelSelectorSchema, "labelSelector is a standard Kubernetes label selector."),
+			"namespaceSelector":  *propsWithDescription(&labelSelectorSchema, "`namespaceSelector` is a label selector against an object's containing namespace, or against namespace objects themselves."),
```

Comment:
"or the object itself, if the object is a namespace"

Basically, make clear that applying the label selector to the object directly only happens if the object is a namespace object.

Reply:
So would the whole thing be:

```
`namespaceSelector` is a standard Kubernetes namespace selector. If defined, make sure to add Namespaces (or the object itself, if that object is a namespace) to your configs.config.gatekeeper.sh object to ensure namespaces are synced into OPA.
```